### ISSUE TITLE
Validating partial documents

### ISSUE THREAD

**Issue by szantner** [cfjedimaster/brackets-w3cvalidation#14]
Hi. Is there a possibility to validate partial documents (for example with the help of comments - like with jslint, jshint)?

As I have seen, it is possible with w3c validator to validate snippets (http://stackoverflow.com/questions/8061087/do-you-know-a-html-snippet-validator)

**Comment by alvint** [comment]
How about doing this the SASS way and assuming that files who's name begin with an underscore are partials? For example "_itemsPanel.html"? And maybe also any HTML file in a folder called "view" or "views".

**Comment by cfjedimaster** [comment]
That feels like a risky thing to assume imo. We could simply say, if no html tag, assume partial. That would still work if folks forget <head> for example but include html, so it would correctly flag it, but if no <html>, then we assume partial. Now - the issue is - does the remote service even support partials or ignoring those rules.

**Comment by cfjedimaster** [comment]
Nope, while the w3c validator on the site lets you use a partial, the API does not: http://validator.w3.org/docs/api.html

So to make this work, I'd need to add a fake <html> wrapper around the content. I could do:

<html><head><title>Temp</title></head><body> content </body></html>

but that may mess up line/character matches in the linter. I don't think this will work well.

**Comment by alvint** [comment]
@cfjedimaster - Good idea. It looks like the validator's web interface supports partials, but possibly only 4.0.1.
